content/en/docs/project-user-guide/image-builder/s2i-templates.md

```diff
@@ -6,4 +6,323 @@ linkTitle: "Customize S2I Templates"
+
+## Prerequisites
+
+S2I template customization can be divided into two parts.
+
+- Part 1: S2I Image Builder customization
+  - assemble (required): the `assemble` script that builds application artifacts from source code.
+  - run (required): the `run` script that executes an application.
+  - save-artifacts (optional): the `save-artifacts` script that manages all dependencies in an incremental building process.
+  - usage (optional): the script that provides instructions.
+  - test (optional): the script for testing.
+- Part 2: definition of S2I template
+
+You need to have the required elements for S2I template customization ready in advance.
+
+{{< notice note >}}
+
+The Image Builder is compatible with that of OpenShift, and you can reuse it in KubeSphere. For more information about S2I Image Builder, refer to [S2IRun](https://github.com/kubesphere/s2irun/blob/master/docs/builder_image.md#s2i-builder-image-requirements).
+
+{{</ notice >}}
+
+## Create an Image Builder
+
+### Step 1: Prepare S2I directory
+
+1. [S2I command line tool](https://github.com/openshift/source-to-image/releases) provides an easy-to-use command to initialize a base directory structure required by the Builder. Run the following commands to install S2I CLI.
+
+   ```bash
+   $ wget https://github.com/openshift/source-to-image/releases/download/v1.1.14/source-to-image-v1.1.14-874754de-linux-386.tar.gz
+   $ tar -xvf source-to-image-v1.1.14-874754de-linux-386.tar.gz
+   $ ls
+   s2i source-to-image-v1.1.14-874754de-linux-386.tar.gz  sti
+   $ cp s2i /usr/local/bin
+   ```
+
+2. This tutorial uses `nginx-centos7` as the name of the Image Builder. Run the `s2i create` command to initialize the base directory structure.
+
+   ```bash
+   s2i create nginx-centos7 s2i-builder-docs
+   ```
+
+3. The directory structure is initialized as follows.
+
+   ```
+   s2i-builder-docs/
+      Dockerfile - a standard Dockerfile to define the Image Builder
+      Makefile - a script for testing and building the Image Builder
+      test/
+         run - a script that runs the application to test whether the Image Builder is working properly
+         test-app/ - directory of the test application
+      s2i/bin
+         assemble - a script that builds the application
+         run - a script that runs the application
+         usage - a script that prints the usage of the Image Builder
+   ```
+
+### Step 2: Modify the Dockerfile
+
+A Dockerfile installs all of the necessary tools and libraries that are needed to build and run an application. This file will also copy the S2I scripts into the output image.
+
+Modify the Dockerfile as follows to define the Image Builder.
+
+#### Dockerfile
+
+```
+# nginx-centos7
+FROM kubespheredev/s2i-base-centos7:1
+
+# Here you can specify the maintainer for the image that you're building
+LABEL maintainer="Runze Xia <runzexia@yunify.com>"
```

Comment:
Use an example name and email address, such as `maintainer name` and `email@xxx.com`.